src/lib/io/sol-iio.c

```diff
@@ -544,12 +544,27 @@ channel_get_pure_name(struct sol_iio_channel *channel)
 {
     size_t channel_name_len;
     char *channel_pure_name;
+    char *channel_name_suffix;
     bool modified = false;
 
     channel_name_len = strlen(channel->name);
-    if (channel_name_len > 2) {
-        char *channel_name_suffix;
+    if (channel_name_len > 5) {
+        channel_name_suffix = channel->name + channel_name_len - 5;
```

Comment:
According to kernel doc, `Documentation/ABI/testing/sysfs-bus-iio`, it's possible to have a channel like `in_intensityY_both_raw`, where Y is an index number. So I'd say that you have to take this into consideration. An idea would be doing what you do now, stripping suffix 'ir' and 'both' (and maybe 'uv') then letting code below process the Y part.
Note that Soletta also has a `strendswith` macro, that may help you to check a string suffix.